Firestore/core/src/credentials/firebase_app_check_credentials_provider_apple.mm

```diff
@@ -31,8 +31,12 @@
 FirebaseAppCheckCredentialsProvider::FirebaseAppCheckCredentialsProvider(
     FIRApp* app, id<FIRAppCheckInterop> app_check) {
   contents_ = std::make_shared<Contents>(app, app_check);
-  std::weak_ptr<Contents> weak_contents = contents_;
 
+  if (app_check == nil) {
```

Comment:
I wonder if it would be better to avoid creating an instance of `FirebaseAppCheckCredentialsProvider` at all if `app_check == nil`. Can `app_check == nil` lead to any unintended behaviour of `FirebaseAppCheckCredentialsProvider` instance?

Reply:
The setup here matches our existing code for FirebaseAuth. The only difference is that the notification key for FirebaseAuth is a constant and always available. I would rather leave this as is.

I am also working on a test now, but my build setup is currently broken so it might take a while.

Reply:
SGTM 👍